### ISSUE TITLE
Handle org.json.JSONObject in saveObjects/ensureObjectID

### ISSUE THREAD

**Issue by Ant-hem** [algolia/algoliasearch-client-java#613]
## Description

@BenoitPerrot & @aseure this issue is concerning our users migrating from Java `v1` to `v3`.
 
To have a straightforward implementation our `v1` users are reusing their `org.json.JSONObject` objects from Java v1, making [ensureObjectID](https://github.com/algolia/algoliasearch-client-java-2/blob/febbaeafd7fbdee4affc9856a4601947a1bd1443/algoliasearch-core/src/main/java/com/algolia/search/util/AlgoliaUtils.java#L32) in `saveObjects` to fail because the method is not expecting that kind of object.

## Solution

- 1st solution would be to add `org.json` dependency and perform the check on the `JSONObject`.  

The weak point of the first solution is we have to add a non-used dependency to the client. In an other hand it looks like the "safer" solution.

- 2nd solution would be to check the type at runtime (without adding `org.json` as dependency) with `this.getClass().getCanonicalName() == org.json.JSONObject` and then perform the extra-check.

The weak point of the second solution is not having the `org.json` dependency will imply tricky code block to handle the all the checks.

- 3rd solution: Work-around: our users can still save those objects with the `batch` method which is not doing an extra check.

WDYT?

**Comment by aseure** [comment]
Thanks for the investigation. If we aim to handle it, I would go for solution 2. if easily doable as I'd rather not add a dependency for such a feature.

**Comment by Ant-hem** [comment]
The 2nd solution can be tricky, I'll PR it in a few days.

**Comment by aseure** [comment]
As discussed IRL, perhaps the best solution would be to not handle this at all. I fear that by handling those JSON object, we may need to handle different ones in the future at the cost of performance. Plus, users migrating to a new major version such as our `v3` should except performance and DX improvements at the cost of changing a bit their code. By having to pass POJO instead of `JSONObject` here, I'm pretty user they will actually be avoiding the extra conversion for every single object they want to send to Algolia.

**TL;DR** Why not **not** supporting this at all in our `v3`?

**Comment by Ant-hem** [comment]
After discussing it within the team, we agreed that supporting `org.json.JSONObject` it's not the philosophy of the `v3` and we would encourage our users to migrate to POJOs. However, our users can still save `JSONObject` with the `batch` method.
